MTA-4637: Pattern matching docs need urgent improvements (quick fix) (#121)
migtools/mta-documentation: docs/topics/yaml-rule-structure-syntax.adoc

    location: "<location>" <2>
    annotated: "<annotated>" <3>
----
<1> A regular expression pattern to match, for example, `org.kubernetes*`.
<2> Specifies the exact location where the pattern needs to be matched, for example, `IMPORT`.

> <3> Checks for specific annotations and their elements, such as name and value in the Java code using a query. For example, the following query matches the Bean(url = “http://www.example.com”) annotation in the method.

Author: ```suggestion
<3> Checks for specific annotations and their elements, such as name and value in the Java code using a query. 
```

* *CONSTRUCTOR_CALL* and *METHOD_CALL*: for matching constructors and methods, respectively. The pattern possibilities are varied, and it is possible to match against specific return types, arguments, and so on.
+
For instance, looking for a method named `method` declared in the `org.konveyor.MyClass` that returns a `List` of a type that extends `java.lang.String` and accepts a single parameter:
+
[source,yaml]
----
java.referenced:
  location: METHOD
  pattern: 'org.konveyor.Myclass.method(*) java.util.List<? extends java.lang.String>'
----

More information about the possibilities of these patterns can be found in the link:[Class SearchPattern section] of the Eclipse documentation, which contain all the required information for building these patterns.

[WARNING]
====
In the current release of {ProductShortName}, fully qualified static method matching does not function as expected.
====

* *TYPE*: matches against types in general, appearing anywhere.
* *INHERITANCE*: matches against a class inheriting from a given type.
* *ANNOTATION*: matches against annotations.
* *IMPLEMENTS_TYPE*: matches against any type implementing the given type.

> * *ENUM_CONSTANT*: matches against enum constants.

Author: ```suggestion
* *ENUM_CONSTANT*: matches against `enum` constants.
```

> * *ENUM_CONSTANT*: matches against enumeration (enum) constants.

Author: ```suggestion
* *ENUM_CONSTANT*: matches against data enumeration (enum) constants.
```

> * *ENUM_CONSTANT*: matches against data enumeration (enum) constants.

Author: ```suggestion
* *ENUM_CONSTANT*: matches against data `enum` constants.
```

* *CONSTRUCTOR_CALL* and *METHOD_CALL*: for matching constructors and methods, respectively. The pattern possibilities are varied, and it is possible to match against specific return types, arguments, and so on.
+
For instance, looking for a method named `method` declared in the `org.konveyor.MyClass` that returns a `List` of a type that extends `java.lang.String` and accepts a single parameter:
+
[source,yaml]
----
java.referenced:
  location: METHOD
  pattern: 'org.konveyor.Myclass.method(*) java.util.List<? extends java.lang.String>'
----

More information about the possibilities of these patterns can be found in the link:[Class SearchPattern section] of the Eclipse documentation, which contain all the required information for building these patterns.

[WARNING]
====
In the current release of {ProductShortName}, fully qualified static method matching does not function as expected.
====

* *TYPE*: matches against types in general, appearing anywhere.
* *INHERITANCE*: matches against a class inheriting from a given type.
* *ANNOTATION*: matches against annotations.
* *IMPLEMENTS_TYPE*: matches against any type implementing the given type.

> * *ENUM_CONSTANT*: matches against enum constants.

Author: ```suggestion
* *ENUM_CONSTANT*: matches against `enum` constants.
```

> <1> `pattern`: A regex pattern to match the desired reference. For example, HttpNotFound.

Author: ```suggestion
<1> `pattern`: A regular expression pattern to match the desired reference. For example, `HttpNotFound`.
```
